arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1936 @@
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/subscriptionId"
+        }, {
+          "name": "resourceGroupName",
+          "in": "path",
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/api-version"
+        }],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/IotHubQuotaMetricInfoListResult"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Devices/checkNameAvailability": {
+      "post": {
+        "tags": ["POST"],
+        "summary": "Check if an IotHub name is available",
+        "description": "Check if an IotHub name is available",
+        "operationId": "IotHubResource_CheckNameAvailability",
+        "consumes": ["application/json", "text/json"],
+        "produces": ["application/json", "text/json"],
+        "parameters": [{
+          "$ref": "#/parameters/subscriptionId",
+          "description": "The subscription identifier."
+        }, {
+          "name": "operationInputs",
+          "in": "body",
+          "description": "The operation inputs. Set only the name parameter in the operationInputs structure to the desired iothub name",
+          "required": true,
+          "schema": {
+            "$ref": "#/definitions/OperationInputs"
+          }
+        }, {
+          "$ref": "#/parameters/api-version"
+        }],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/IotHubNameAvailabilityInfo"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}/listkeys": {
+      "post": {
+        "tags": ["POST"],
+        "summary": "Get all Shared Access Policies for an IotHub",
+        "description": "Get all Shared Access Policies for an IotHub",
+        "operationId": "IotHubResource_ListKeys",
+        "consumes": [],
+        "produces": ["application/json", "text/json"],
+        "parameters": [{
+          "name": "resourceName",
+          "in": "path",
+          "description": "Name of the resource.",
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/subscriptionId"
+        }, {
+          "name": "resourceGroupName",
```

Comment:
Missing description. (Descriptions are important, they are used to generate the help documentation for the APIs)